### PR TITLE
updated tests to clear up confusion

### PR DESCRIPTION
**Pull request by ndalcin**
Some students were having trouble reading the tests regarding changing happiness and hygiene, so I updated those tests to make it easier to understand.I also added the specific error that should be raised when using the raise_error matcher to remove warning message from terminal ( see Issue #13 )

### CODE REVIEW

**Comment by morgvanny** [comment]
it feels like more of a problem with those students’ understanding of what += is actually doing than it is a problem with the tests, but I’m not particularly opposed to the change either, if it helps eliminate confusion. I like that the specific error error was added to be raised

**Comment by maxcell** [comment]
So I had exactly the same thoughts as far as @ndalcin that the tests have to be changed. And my initial thought this week was that we should change it to be something like `person.happiness = person.happiness + 100` or something of that sorts (i.e. writing the longhand). 

And after reviewing the change, **I am 100% onboard with @ndalcin** and this is because our method is supposed to only test a single thing here. It is should really see if we can change our happiness/hygiene and that's it. When we 1) combine the reader and writer, this actually makes the test much more complex, 2) us using shorthand when a student is just first starting shouldn't be our default regardless of if they understand it or not. We should assume being a beginner always and make our tests more readable. The reason we also have no reason to do this is because we are testing the logic of `+=`/`-=` inside of `take_bath`/`work_out`. It makes sense for us to separate our logic out from the tests because consistently students don't understand that they can write the custom logic of setting the maximum inside of the writer method they create and I think that's because the test wasn't clear.

TL;DR I'm here for it all!

File: spec/person_spec.rb [hunk 1]
```diff
@@ -203,144 +194,140 @@
...
-      context "topic: other" do
+      context 'topic: other' do
         it "returns 'blah blah blah blah blah' if the topic is not politics or weather" do
-          convo = "blah blah blah blah blah"
-          expect(penelope.start_conversation(felix, "programming")).to eq(convo)
+          convo = 'blah blah blah blah blah'
+          expect(penelope.start_conversation(felix, 'programming')).to eq(convo)
         end
 
         it "if topic is not politics or weather, it doesn't affect happiness" do
```
Comment (maxcell) [inline spec/person_spec.rb]: Nitpick: Change this to single quotes

File: spec/person_spec.rb [hunk 2]
```diff
@@ -203,144 +194,140 @@
...
             expect(person).to receive(:happiness=).with(11).and_return(10)
           end
-          penelope.start_conversation(felix, "weather")
+          penelope.start_conversation(felix, 'weather')
         end
       end
 
-      context "topic: other" do
+      context 'topic: other' do
         it "returns 'blah blah blah blah blah' if the topic is not politics or weather" do
```
Comment (maxcell) [inline spec/person_spec.rb]: Nitpick: Change this to single quotes

File: spec/person_spec.rb [hunk 3]
```diff
@@ -203,144 +194,140 @@
...
-          people.each  { |p| p.happiness = 7 }
-          penelope.start_conversation(felix, "weather")
-          people.each  { |p| expect(p.happiness).to eq(8) }
+        it 'if topic is weather, it makes both people in the convo one point more happy' do
+          people.each { |p| p.happiness = 7 }
+          penelope.start_conversation(felix, 'weather')
+          people.each { |p| expect(p.happiness).to eq(8) }
         end
 
         it "never makes either party more than 10 points happy
```
Comment (maxcell) [inline spec/person_spec.rb]: Nitpick: Change this to single quotes

File: spec/person_spec.rb [hunk 4]
```diff
@@ -203,144 +194,140 @@
...
-          penelope.start_conversation(felix, "politics")
+          penelope.start_conversation(felix, 'politics')
         end
       end
 
-
-      context "topic: weather" do
-
+      context 'topic: weather' do
         it "returns 'blah blah sun blah rain' if the topic is weather" do
```
Comment (maxcell) [inline spec/person_spec.rb]: Nitpick: Change this to single quotes

File: spec/person_spec.rb [hunk 5]
```diff
@@ -203,144 +194,140 @@
...
-          people.each  { |person| person.happiness = 7 }
-          penelope.start_conversation(felix, "politics")
-          people.each  { |person| expect(person.happiness).to eq(5) }
+        it 'if topic is politics, it makes both people in the convo two points less happy' do
+          people.each { |person| person.happiness = 7 }
+          penelope.start_conversation(felix, 'politics')
+          people.each { |person| expect(person.happiness).to eq(5) }
         end
 
         it "never makes either party less than 0 points happy (never negative)
```
Comment (maxcell) [inline spec/person_spec.rb]: Nitpick: Change this to single quotes

File: spec/person_spec.rb [hunk 6]
```diff
@@ -203,144 +194,140 @@
...
-        expect { penelope.start_conversation(felix, "politics") }.to_not raise_error
+    describe '#start_conversation' do
+      it 'accepts two arguments, a person to strike a convo with and a topic' do
+        expect { penelope.start_conversation(felix, 'politics') }.to_not raise_error
       end
 
-      context "topic: politics" do
-
+      context 'topic: politics' do
         it "returns 'blah blah partisan blah lobbyist' if the topic is politics" do
```
Comment (maxcell) [inline spec/person_spec.rb]: Nitpick: Change this to single quotes

File: spec/person_spec.rb [hunk 7]
```diff
@@ -203,144 +194,140 @@
...
-        [felix, penelope].each {|person| person.happiness = 9 }
+      it 'calls on the #happiness= method to increment happiness' do
+        [felix, penelope].each { |person| person.happiness = 9 }
         people.each do |person|
           expect(person).to receive(:happiness=).with(12).and_return(10)
         end
         penelope.call_friend(felix)
       end
 
       it "returns a string that reflects the caller's side of the conversation" do
```
Comment (maxcell) [inline spec/person_spec.rb]: Nitpick: Change this to single quotes

File: spec/person_spec.rb [hunk 8]
```diff
@@ -203,144 +194,140 @@
...
       end
 
-      it "makes the friend happier by three points" do
+      it 'makes the friend happier by three points' do
         felix.happiness = 5
         penelope.call_friend(felix)
         expect(felix.happiness).to eq(8)
       end
 
       it "never makes the happiness of either party exceed 10
```
Comment (maxcell) [inline spec/person_spec.rb]: Nitpick: Change this to single quotes

File: spec/person_spec.rb [hunk 9]
```diff
@@ -203,144 +194,140 @@
         expect(penelope.happiness).to eq(10)
```
Comment (maxcell) [inline spec/person_spec.rb]: Nitpick: Change this to single quotes (Line 189)

File: spec/person_spec.rb [hunk 10]
```diff
@@ -183,13 +174,13 @@
         expect(penelope.hygiene).to eq(0)
```
Comment (maxcell) [inline spec/person_spec.rb]: Nitpick: Change this to single quotes (Line 169)

File: spec/person_spec.rb [hunk 11]
```diff
@@ -142,16 +134,16 @@
...
       end
 
-      it "returns a song" do
+      it 'returns a song' do
         penelope.hygiene = 5
-        expect(penelope.take_bath).to eq("♪ Rub-a-dub just relaxing in the tub ♫")
+        expect(penelope.take_bath).to eq('♪ Rub-a-dub just relaxing in the tub ♫')
       end
 
       it "can't make a person cleaner by 10 points (hint: use the custom #hygiene= method)" do
```
Comment (maxcell) [inline spec/person_spec.rb]: Nitpick: Change this to single quotes

File: spec/person_spec.rb [hunk 12]
```diff
@@ -142,16 +134,16 @@
       end
```
Comment (maxcell) [inline spec/person_spec.rb]: Nitpick: Change this to single quotes

File: spec/person_spec.rb [hunk 13]
```diff
@@ -1,137 +1,129 @@
...
           people.each do |person|
-            original_hygiene = person.hygiene
-            person.hygiene += 1
-            expect(person.hygiene).to eq(original_hygiene + 1)
+            person.hygiene = 9
+            expect(person.hygiene).to eq(9)
           end
         end
 
         it "a person instance's hygiene index cannot exceed 10 " do
```
Comment (maxcell) [inline spec/person_spec.rb]: Nitpick: Change this to single quotes

File: spec/person_spec.rb [hunk 14]
```diff
@@ -1,137 +1,129 @@
...
         it "a person's happiness doesn't exceed 10" do
           people.each do |person|
-            original_happiness = person.happiness
-            person.happiness += 100
+            person.happiness = 11
             expect(person.happiness).to eq(10)
           end
         end
 
         it "a person's happiness doesn't go below 0" do
```
Comment (maxcell) [inline spec/person_spec.rb]: Nitpick: Change this to single quotes

File: spec/person_spec.rb [hunk 15]
```diff
@@ -1,137 +1,129 @@
...
           people.each do |person|
-            original_happiness = person.happiness
-            person.happiness += 1
-            expect(person.happiness).to eq(original_happiness + 1)
+            person.happiness = 9
+            expect(person.happiness).to eq(9)
           end
         end
 
         it "a person's happiness doesn't exceed 10" do
```
Comment (maxcell) [inline spec/person_spec.rb]: Nitpick: Change this to single quotes

**Comment by morgvanny** [comment]
@maxcell are we suddenly taking a hard stance on single vs double quotes in ruby? because just a very quick glance at our readmes, and review videos, and we are using double quotes all over the place all the time. It's been argued for years that it's more pragmatic particularly for beginners to just prefer double quotes, since they pretty much always work (except for quotes within quotes). Overall the consensus has always seemed to me to be that it's not really worth caring much about

**Comment by maxcell** [comment]
@morgvanny I think we shouldn't change them. But if we do, we should change every occurrence. That's the purpose of the nitpick. I think for this PR we should simply undo the change for the quotes and have a separate one and decide if this is something we want.

**Comment by morgvanny** [comment]
@maxcell I see what you mean. I think some people have auto-formatting that enforces single quotes, which is what I'm guessing happened here
